File: src/practice/luogu/p5707/Main.py
# https://www.luogu.com.cn/problem/P5707

import math

ss = input().split()
s = int(ss[0])
v = int(ss[1])
time = 10 + math.ceil(s / v)
h = 7
m = 60
h -= int(time / 60)
m -= time % 60
while h < 0:
    h += 24
print(('0' + str(h) if h < 10 else str(h)) + ':' +
      ('0' + str(m) if m < 10 else str(m)))

File: src/practice/luogu/p5707/Main.kt
// https://www.luogu.com.cn/problem/P5707
fun main() {
  val i = readLine()!!.split(" ").map { it.toInt() }
  val t: Int = 10 + Math.ceil(1.0 * i[0] / i[1]).toInt()
  var hr = 7 - t / 60
  val min = 60 - t % 60
  if (hr < 0) hr += 24
  print(if (hr < 10) "0" + hr else hr)
  println(":" + (if (min < 10) "0" + min else min))
}
